examples/golang/waku.go

```diff
@@ -0,0 +1,326 @@
+    return contentTopic
+}
+
+func WakuPubsubTopic(ctx unsafe.Pointer, topicName string) string {
+	var cTopicName = C.CString(topicName)
+	var str = C.allocMyString()
+
+	defer C.free(unsafe.Pointer(cTopicName))
+	defer C.freeMyString(str)
+
+	C.cGoWakuPubsubTopic(ctx, cTopicName, str)
+
+	var pubsubTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return pubsubTopic
+}
+
+func WakuDefaultPubsubTopic(ctx unsafe.Pointer) string {
+	var str = C.allocMyString()
+	defer C.freeMyString(str)
+	C.cGoWakuDefaultPubsubTopic(ctx, str)
+	var defaultPubsubTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return defaultPubsubTopic
+}
+
+func WakuRelayPublish(ctx unsafe.Pointer,
+					pubsubTopic string,
+					message string,
+					timeoutMs int) {
+	var cPubsubTopic = C.CString(pubsubTopic)
+	var msg = C.CString(message)
+
+	defer C.free(unsafe.Pointer(cPubsubTopic))
+	defer C.free(unsafe.Pointer(msg))
+
+	C.cGoWakuRelayPublish(ctx, cPubsubTopic, msg, C.int(timeoutMs))
+}
+
+func WakuRelaySubscribe(ctx unsafe.Pointer, pubsubTopic string) {
+	var cPubsubTopic = C.CString(pubsubTopic)
+	defer C.free(unsafe.Pointer(cPubsubTopic))
+	C.cGoWakuRelaySubscribe(ctx, cPubsubTopic)
+}
+
+func WakuRelayUnsubscribe(ctx unsafe.Pointer, pubsubTopic string) {
+	var cPubsubTopic = C.CString(pubsubTopic)
+	defer C.free(unsafe.Pointer(cPubsubTopic))
+	C.cGoWakuRelayUnsubscribe(ctx, cPubsubTopic)
+}
+
+func WakuConnect(ctx unsafe.Pointer, peerMultiAddr string, timeoutMs int) {
+	var cPeerMultiAddr = C.CString(peerMultiAddr)
+	defer C.free(unsafe.Pointer(cPeerMultiAddr))
+
+	C.cGoWakuConnect(ctx, cPeerMultiAddr, C.int(timeoutMs))
+}
+
+func WakuListenAddresses(ctx unsafe.Pointer) string {
+	var str = C.allocMyString()
+	defer C.freeMyString(str)
+	C.cGoWakuListenAddresses(ctx, str)
+	var listenAddresses = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return listenAddresses
+}
+
+func main() {
+	config := `{
+		"host": "0.0.0.0",
+		"port": 30304,
+		"key": "11d0dcea28e86f81937a3bd1163473c7fbc0a0db54fd72914849bc47bdf78710",
+		"relay": true
+	}`
+
+	ctx := WakuNew(config)
+	WakuSetEventCallback(ctx)
+	WakuRelaySubscribe(ctx, WakuDefaultPubsubTopic(ctx))
+	WakuConnect(ctx,
+		// tries to connect to a localhost node with key: 0d714a1fada214dead6dc9c7274585eca0ff292451866e7d6d677dc818e8ccd2
+		"/ip4/0.0.0.0/tcp/60000/p2p/16Uiu2HAmVFXtAfSj4EiR7mL2KvL4EE2wztuQgUSBoj2Jx2KeXFLN",
+		10000)
+	WakuStart(ctx)
+
+	fmt.Println("Version:", WakuVersion(ctx))
+	fmt.Println("Custom content topic:", WakuContentTopic(ctx, "appName", 1, "cTopicName", "enc"))
+	fmt.Println("Custom pubsub topic:", WakuPubsubTopic(ctx, "my-ctopic"))
+	fmt.Println("Default pubsub topic:", WakuDefaultPubsubTopic(ctx))
+	fmt.Println("Listen addresses:", WakuListenAddresses(ctx))
+	
+	for {
+		// A simple wait to let the Waku node to run.
+		// Notice that the Waku node runs in a separate thread.
+		time.Sleep(time.Second)
+	}
+}
```

Comment:
Even if the node will stop anyway once the program exits, it is probably a good idea to manually stop the node, so devs see that the function exists!

```suggestion
  WakuStop()
}
```